File: go/internal/app/bitmex/models/trade_bin.go
package models

import "time"

// TradeBin trade bin
type TradeBin struct {
	Close           float64   `json:"close,omitempty"`
	ForeignNotional float64   `json:"foreignNotional,omitempty"`
	High            float64   `json:"high,omitempty"`
	HomeNotional    float64   `json:"homeNotional,omitempty"`
	LastSize        int64     `json:"lastSize,omitempty"`
	Low             float64   `json:"low,omitempty"`
	Open            float64   `json:"open,omitempty"`
	Symbol          string    `json:"symbol"`
	Timestamp       time.Time `json:"timestamp"`
	Trades          int64     `json:"trades,omitempty"`
	Turnover        int64     `json:"turnover,omitempty"`
	Volume          int64     `json:"volume,omitempty"`
	Vwap            float64   `json:"vwap,omitempty"`
}

File: ts/src/internal/app/bitmex/models/trade_bin.ts
// TradeBin trade bin
export default interface TradeBin {
  close: number;
  foreignNotional: number;
  high: number;
  homeNotional: number;
  lastSize: number;
  low: number;
  open: number;
  symbol: string;
  timestamp: Date;
  trades: number;
  turnover: number;
  volume: number;
  vwap: number;
}
